Slim/Control/Queries.pm

```diff
@@ -4636,26 +4689,41 @@ sub worksQuery {
 		my $sth = $dbh->prepare_cached($sql);
 		$sth->execute( @{$p} );
 
-		my ($work, $workId, $composer, $composerId, $nameSort, $titleSort );
-		$sth->bind_columns(\$work, \$workId, \$composer, \$composerId, \$nameSort, \$titleSort);
+		my ($work, $workId, $composer, $composerId, $nameSort, $titleSort, $covers, $album_ids);
+		$sth->bind_columns(\$work, \$workId, \$composer, \$composerId, \$nameSort, \$titleSort, \$covers, \$album_ids);
 
 		while ( $sth->fetch ) {
-			#$id += 0;
+
+			my @allImages = split(/,/,$covers);
+			my $image = @allImages[0];
+			my $images = scalar @allImages > 4 ? join(',', @allImages[0..3]) : join(',',@allImages);
+			
+
 			utf8::decode($work) if $work;
 			utf8::decode($composer) if $composer;
-			$request->addResultLoop($loopname, $chunkCount, 'work', $work);
 			$request->addResultLoop($loopname, $chunkCount, 'work_id', $workId);
-			$request->addResultLoop($loopname, $chunkCount, 'composer', $composer);
+			if ( $composerCount == 1 && $composerId eq $artistID ) {
+				$request->addResultLoop($loopname, $chunkCount, 'composer', $work);
+				$request->addResultLoop($loopname, $chunkCount, 'work', " ");
+			} else {
+				$request->addResultLoop($loopname, $chunkCount, 'composer', $composer);
+				$request->addResultLoop($loopname, $chunkCount, 'work', $work);
+			}
```

Comment:
I'm not sure I understand this: are you returning the work name as the composer? That sounds wrong to me.

Reply:
It's really line1/line2 - the idea is that if displaying works for a composer such that all works have the same composer, we suppress the composer name (it was a user request). 

I'll look at renaming to line1 and line2 to avoid this confusion.

Reply:
Thanks for the confirmation 😀. "line" is a UI concept, not data 😉. Please do not include lines here. Really. That's UI code.

Reply:
So leave it as it is, or pass a flag to tell the UI that all results are composed by the browsed/searched-for artist, and have Material and BrowseLibrary/Works.pm do the formatting separately?